dev/com.ibm.ws.springboot.utility/src/com/ibm/ws/springboot/utility/tasks/ThinAppTask.java

```diff
@@ -0,0 +1,192 @@
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.springboot.utility.tasks;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import com.ibm.ws.app.manager.springboot.internal.SpringConstants;
+import com.ibm.ws.app.manager.springboot.util.SpringBootThinUtil;
+import com.ibm.ws.springboot.utility.IFileUtility;
+import com.ibm.ws.springboot.utility.SpringBootUtilityReturnCodes;
+import com.ibm.ws.springboot.utility.utils.ConsoleWrapper;
+
+/**
+ *
+ */
+public class ThinAppTask extends BaseCommandTask {
+    static final String SLASH = String.valueOf(File.separatorChar);
+
+    static final String ARG_SOURCE_APP = "--sourceAppPath";
+    static final String ARG_DEST_THIN_APP = "--targetThinAppPath";
+    static final String ARG_DEST_LIB_CACHE = "--targetLibCachePath";
+    static final String ARG_PARENT_LIB_CACHE = "--parentLibCachePath";
+    static final Set<String> knownArguments = new HashSet<>(Arrays.asList(ARG_SOURCE_APP, ARG_DEST_THIN_APP, ARG_DEST_LIB_CACHE, ARG_PARENT_LIB_CACHE));
+
+    private final IFileUtility fileUtility;
+    protected ConsoleWrapper stdin;
+    protected PrintStream stdout;
+    protected PrintStream stderr;
+
+    public ThinAppTask(IFileUtility fileUtility, String scriptName) {
+        super(scriptName);
+        this.fileUtility = fileUtility;
+
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public String getTaskName() {
+        return "thin";
+    }
+
+    @Override
+    public String getTaskDescription() {
+        return getOption("thin.desc", true);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public String getTaskHelp() {
+        return getTaskHelp("thin.desc", "thin.usage.options",
+                           "thin.required-key.", "thin.required-desc.",
+                           "thin.optional-key.", "thin.optional-desc.",
+                           scriptName);
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @return
+     */
+    @Override
+    public SpringBootUtilityReturnCodes handleTask(ConsoleWrapper stdin, PrintStream stdout, PrintStream stderr, String[] args) throws Exception {
+        this.stdin = stdin;
+        this.stdout = stdout;
+        this.stderr = stderr;
+
+        validateArgumentList(args);
+
+        String sourceAppPath = getArgumentValue(ARG_SOURCE_APP, args);
+        String destThinAppPath = getArgumentValue(ARG_DEST_THIN_APP, args);
+        String destLibCachePath = getArgumentValue(ARG_DEST_LIB_CACHE, args);
+        String parentLibCachePath = getArgumentValue(ARG_PARENT_LIB_CACHE, args);
+
+        if (!fileUtility.isFile(sourceAppPath)) {
+            stdout.println(getMessage("thin.abort"));
```

Comment:
should errors here be output to stderr? Seems you do that elsewhere in the utility